### PR TITLE
New blocks

### PR DESCRIPTION
**Pull request by Git-the-Sanz**
#### How to test it?

[Workspace](https://sanz--powerplanet.myvtex.com/)

#### Related to / Depends on

This branch has [this pr](https://github.com/vtex-apps/binding-selector/pull/12) already merged

#### Screenshots or example usage:

The action bar implementing the new selector type
Action bar

The new current binding block being used as a modal trigger to open a modal with the new binding list showcasing the combined prop
Current binding with modal

The new binding list showcasing the flag prop
Binding list

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,89 +1,119 @@
...
+www.testperformancev.com/pt/iphone-11-128gb-vermelho/p -> Canonical
+www.testperformancev.com/es/iphone-11-128gb-rojo/p -> Alternate
 
-It has the possibility to update the Session and Cart's Sales Channel, this changes the prices and stock available for each store. It is up to the user to toggle this setting ON of OFF inside the App's settings. 
+It has the possibility to update the Session and Cart's Sales Channel, this changes the prices and stock available for each store. It is up to the user to toggle this setting ON of OFF inside the App's settings.
 
-The App is geared towards an improvement on Site Experience for the customer and it will have a direct impact on the store's conversion.  
+The App is geared towards an improvement on Site Experience for the customer and it will have a direct impact on the store's conversion.
 
 ![bindingfunctionality](/docs/bindingfunctionality.gif)
```
Comment (filafb) [inline docs/README.md:22]: It would be nice to have the gif with the new version.

File: react/BindingChallenge.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  const {
+    data: { currentBinding, bindingsError, loadingBindings },
+  } = useBinding()
+
+  useEffect(() => {
+    fetch('/_v/binding-selector/viewer-country').then((response) =>
+      response.text().then((countryISO: string | null) => {
+        !countryISO || setViewerCountry(countryISO)
+      })
+    )
```
Comment (filafb) [inline react/BindingChallenge.tsx]: Not sure how you feel about that, but I would prefer keeping the same style we are resolving promises throughout the code - with `async / await`. I know the pattern inside `useEffect` can be weird, so just want to hear your thoughts.

File: react/BindingChallenge.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    const bindingLocale = currentBinding.defaultLocale
+      ?.substr(3, 2)
+      .toUpperCase()
+
+    if (viewerCountry === bindingLocale) {
+      return
+    }
+
+    setDisplay(true)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (filafb) [inline react/BindingChallenge.tsx]: I think we can remove this eslint flag. I guess all variables are listed in the deps.

**Review by filafb — CHANGES_REQUESTED** [review]
Nice. Just a couple of comments, nothing a big deal.

I would change the base to be Edy's PR instead of main. This way we can have his branch as the `next release` one for this new implementation.

And since Ametller is using the app, it's important to note that until they can change the flags via admin, this version will show two Spanish flags if they try to use the new layout.

File: docs/README.md (hunk 1 above)
Comment (Git-the-Sanz) [reply docs/README.md:22]: Yeah! Will do when we finish everything

File: react/BindingChallenge.tsx (hunk 3 above)
Comment (Git-the-Sanz) [reply react/BindingChallenge.tsx]: Removed

File: react/BindingChallenge.tsx (hunk 2 above)
Comment (Git-the-Sanz) [reply react/BindingChallenge.tsx]: Changed

**Pull request merged**